### PR TITLE
Shahzad - Extracting scanreport values for scan_report and scan_report_table

### PR DESCRIPTION
**Pull request by shahzadmumtaz22**
Addition of two new endpoints to extract all scanreportvalues for scan_report (requirement from Simon Tarr) and scan_report_table (requirement from Calum).  

1. Returning all scanreport values for a scan_report - http://localhost:8080/api/scanreportvaluesfilterscanreport/?scan_report=651
2. Returning all scanreport values for a scan_report_table - http://localhost:8080/api/scanreportvaluesfilterscanreporttable/?scan_report_table=8

### CODE REVIEW

**Comment by calmacx** [comment]
hi @shahzadmumtaz22 can i use a filter with the api call:
```
/api/scanreportfieldsfilter/?scan_report_table={_id}&fields=name
```

For each scan report field, I dont need to return all the information

E.g. it is returning to me:
```
{'id': 9156, 'name': 'EAVE_LINKNO', 'description_column': '', 'created_at': '2021-06-21T12:51:38.254476Z', 'updated_at': '2021-06-21T12:51:38.260808Z', 'type_column': 'VARCHAR', 'max_length': 11, 'nrows': -1, 'nrows_checked': 100, 'fraction_empty': '0.00', 'nunique_values': 62, 'fraction_unique': '0.62', 'ignore_column': None, 'flag_column': '', 'is_patient_id': False, 'is_date_event': False, 'is_birth_date': False, 'is_ignore': False, 'classification_system': '', 'pass_from_source': True, 'date_type': '', 'concept_id': -1, 'field_description': None, 'scan_report_table': 424}
```
but i only need:
```
{'id': 9156, 'name': 'EAVE_LINKNO'}
```

How can we turn on a filter as we did for:
```
/api/scanreportvaluesfilterscanreporttable/?scan_report_table={_id}&fields=value,frequency,scan_report_field
```
?

**Comment by shahzadmumtaz22** [comment]
Hi Calum, I have made changes and now you can access the endpoint to return only specific values from the scan_report_field table. The URL something like this http://localhost:8080/api/scanreportfieldsfilter/?scan_report_table=8&fields=name,description_column

**Comment by shahzadmumtaz22** [comment]
The conflicts were not there when I made this PR earlier today and I think a day before yesterday myself and simon resolved an issue that was to return all scan report values for a scan report and I believe he made a PR and that was merged earlier on (around 2 hours before) and is causing a conflict . I think we need to resolve this tomorrow because his naming convention is different than mine or as a quick fix I should remove my code (though that will create an inconsistency in terms of naming convention and parameter passing). Lets discuss this tomorrow in daily stand up.

File: api/mapping/views.py [hunk 1]
```diff
@@ -307,7 +307,26 @@ class ScanReportValueFilterViewSet(viewsets.ModelViewSet):
     queryset=ScanReportValue.objects.all()
     serializer_class=ScanReportValueSerializer
     filter_backends=[DjangoFilterBackend]
-    filterset_fields=['scan_report_field', 'scan_report_field__scan_report_table__scan_report', 'value']    
+    filterset_fields=['scan_report_field', 'value']
```
[the remark this answers is not in the document]
Comment (calmacx) [reply api/mapping/views.py:310]: actually @shahzadmumtaz22 is this fine?

[the remark this answers is not in the document]
Comment (shahzadmumtaz22) [reply api/mapping/views.py:310]: Hi Calum, this is fine. I have double checked it. But once you will merge, then just let me know, I will check that in the devel as well.

**Pull request merged**